### PR TITLE
Update operational credentials code to match the specifications

### PR DESCRIPTION
**Pull request by pan-apple**
#### Problem
Need updates to operational credentials provisioning code to match the latest specifications.

#### Change overview
1. Remove trusted-root cluster
2. Move commands from trusted-root cluster to operational-credentials-cluster
3. Regenerate the ZAP code
4. Update provisioning flow to first provision root certificate, followed by operational certificates
5. Add support for provisioning ICA certificates, if available. Use new API to generate certificate array and send the array to the device.

#### Testing
This change impacts the device commissioning flow.
Tested commissioning using Python controller app, chip-tool, and iOS CHIPTool app.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @jelderton @Damian-Nordic @LuDuda ?

**Comment by pan-apple** [comment]
rebased

File: src/transport/AdminPairingTable.h [hunk 1]
```diff
@@ -194,8 +194,7 @@ class DLL_EXPORT AdminPairingInfo
 
         Crypto::P256SerializedKeypair mOperationalKey;
         uint8_t mRootCert[kMaxChipCertSize];
-        uint8_t mOperationalCert[kMaxChipCertSize];
-
+        uint8_t mOperationalCert[kMaxChipCertSize * 2];
```
Comment (mspang) [inline src/transport/AdminPairingTable.h:197]: Why was this needed?

Comment (pan-apple) [reply src/transport/AdminPairingTable.h:197]: We are storing an array of certs now, NOC and ICA certs. The CPP file has the comment that reflects it.

Comment (pan-apple) [reply src/transport/AdminPairingTable.h:197]: I'll add the comment here as well.

**Comment by pan-apple** [comment]
rebased

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/zap-templates/zcl/operational-credentials-cluster.xml [hunk 2]
```diff
@@ -81,8 +81,7 @@ fabric-scoped data.</description>
 
     <command source="client" code="0x06" name="AddOpCert" optional="false">
       <description>Sender is requesting to add the new node operational certificate.</description>
-      <arg name="NOC" type="OCTET_STRING"/>
-      <arg name="ICACertificate" type="OCTET_STRING"/>
+      <arg name="OperationalCert" type="OCTET_STRING" array="true"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/operational-credentials-cluster.xml:84]: Why `array="true"` here?  This is a single OCTET_STRING, not an array of them, right?

Comment (pan-apple) [reply src/app/zap-templates/zcl/operational-credentials-cluster.xml:84]: It's expected to be an array of certs. Not sure if `array="true"` is actually helping with anything here.

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/operational-credentials-cluster.xml:84]: The other places I see `array="true"` in our XML are things like:
```
<arg name="protocolsSupported" type="OTADownloadProtocol" array="true"/>
```
which in the spec has type `list[DownloadProtocolEnum]`.  And the codegen for that generates `uint8_t *`, not `uint8_t`....

Comment (pan-apple) [reply src/app/zap-templates/zcl/operational-credentials-cluster.xml]: Removed the `array="true"`

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/operational-credentials-cluster.xml]: Per offline discussion: this should not have `array="true"` and `array="true"` is kinda broken for OCTET_STRING...

**Review by tcarmelveilleux — APPROVED** [review]
There are many steps that are starting to line-up to spec, so this is great to see!

However, note that the operational certificate provisioning is still not "to spec", due to the NodeID being locally generated, and the full CSR not being passed along with signature and DA keys such that a CA could actually trust it.

**Comment by pan-apple** [comment]
> There are many steps that are starting to line-up to spec, so this is great to see!
> 
> However, note that the operational certificate provisioning is still not "to spec", due to the NodeID being locally generated, and the full CSR not being passed along with signature and DA keys such that a CA could actually trust it.

The node ID is assigned by the commissioner app right now. I don't think it's being locally generated.
DA keys and signature verification depend on device attestation implementation. I understand that attestation happens before operational credentials provisioning as per spec. But until that feature is implemented there's no way to check the signature of OpCSR.
